src/Generators/ModelGenerator.php

```diff
@@ -125,17 +125,18 @@ protected function buildClassPhpDoc(Model $model)
                 $phpDoc .= PHP_EOL;
                 $phpDoc .= ' * @property string|null $' . $column->name() . '_type';
                 $phpDoc .= PHP_EOL;
+            } elseif (
+                $model->usesSoftDeletes() &&
+                $model->softDeletesDataType() === mb_strtolower($column->dataType())
```

Comment:
```php
$model->usesSoftDeletes()  // returns true if  model uses SoftDeletes.
$model->softDeletesDataType() // returns (SoftDeletes|SoftDeletesTz)
$column->dataType() // returns (SoftDeletes|SoftDeletesTz)
```
Since a `$column->dataType()` column has 2 dataTypes `SoftDeletes` or `SoftDeletesTz` we want to make sure it matches the one the user provided in the draft file `$model->softDeletesDataType()`, but only if `$model->usesSoftDeletes() ` is true.

Reply:
I'm reluctant to add any new code that is not needed for this PR, or at least not proven by a test.

Reply:
I understand. you are welcome to make any changes you think are necessary. 

- [x] Allow edits by maintainers